### ISSUE TITLE
message send from device 

### ISSUE THREAD

**Issue by Rhaydae** [tananaev/traccar#3047]
Hi tananaev,
I want to ask you where can i find  the message send from the device in binary format before any change in the content of the message

**Comment by tananaev** [comment]
In the logs (tracker-server.log).

**Comment by Rhaydae** [comment]
Ok thanks for your answer, but please  I want to find it in the source code because I want to treat this message

**Comment by tananaev** [comment]
What exactly do you want to find in the source code?

**Comment by Rhaydae** [comment]
the original message sent from device before any conversion

**Comment by tananaev** [comment]
Depending on the protocol, you can usually find it in the protocol decoder and/or frame decoder.

**Comment by Rhaydae** [comment]
yes I find it in decoder but it seems like is not complete ,its not like the message writen in the file tracker-ser.log

**Comment by tananaev** [comment]
You should probably be more specific about which class you are looking at.
